### PR TITLE
Sigv4a

### PR DESCRIPTION
**Pull request by bretambrose**
* Sigv4a implementation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,18 +3,19 @@ name: CI
...
-      - '*'
-      - '!main'
+      - "*"
+      - "!main"
 
 env:
   BUILDER_VERSION: v0.8.11
   BUILDER_SOURCE: releases
   BUILDER_HOST: https://d19elf31gohf1l.cloudfront.net
-  PACKAGE_NAME: aws-crt-nodejs
```
Comment (TingDaoK) [inline .github/workflows/ci.yml:13]: hmmm, I think I have the branch that only has the sigv4a stuff in it and those CI related stuff was removed [here](https://github.com/awslabs/aws-crt-nodejs-staging/pull/5/files). So, I guess we can squash that PR and cherry-pick the squashed commit.

Comment (TingDaoK) [reply .github/workflows/ci.yml:13]: Or just remove all those changes in ci.yml, they are not needed for public repo. Either way.

**Review by justinboswell — APPROVED** [review]

File: source/module.c [hunk 2]
```diff
@@ -368,6 +368,8 @@ napi_status aws_napi_dispatch_threadsafe_function(
     });
     /* Must always decrement the ref count, or the function will be pinned */
     napi_status release_status = napi_release_threadsafe_function(tsfn, napi_tsfn_release);
+    /* main thread can exit now */
+    AWS_NAPI_ENSURE(env, napi_unref_threadsafe_function(env, tsfn));
```
Comment (TingDaoK) [inline source/module.c:372]: This is not needed. It came from some testing I had from staging repo that I tired to fix the crash about threadsafe function. The public repo already have a better fix for the crash. Seems like I forgot to remove it from staging repo.

**Review by TingDaoK — APPROVED** [review]
Thanks!

**Pull request merged**